kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/NetworkConfigurationServiceImpl.java

```diff
@@ -749,6 +752,16 @@ private void getWifiDefinition(NetInterfaceType type, ObjectFactory objectFactor
         }
     }
 
+    private static Map<String, Object> discardModifiedNetworkInterfaces(final Map<String, Object> properties) {
+        if (!properties.containsKey(MODIFIED_INTERFACE_NAMES)) {
+            return properties;
+        }
+
+        final Map<String, Object> result = new HashMap<>(properties);
+        result.remove(MODIFIED_INTERFACE_NAMES);
+        return result;
```

Comment:
I think you should be able to just use the `remove` without checking if the Map contains the key since, [from the documentation](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#remove-java.lang.Object-):

> Removes the mapping for the specified key from this map **if present**.

So it should simplify into:

```suggestion
        final Map<String, Object> result = new HashMap<>(properties);
        result.remove(MODIFIED_INTERFACE_NAMES);
        return result;
```

or whatever you like...

It can be done directly on line 200 (see other comment)

Reply:
I created the method in this way so that it will be able to handle immutable maps as well, not all Java Map implementation supports the remove operation: https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#remove-java.lang.Object. If they don't, calling remove will throw an [UnsupportedOperationException](https://docs.oracle.com/javase/8/docs/api/java/lang/UnsupportedOperationException.html).

I'm pretty sure that the map that we receive on the activate() and update() methods is immutable.

Probably in the code as it is now it will be always be applied to mutable maps, but I've kept it in this way just to stay on the safe side in case of a refactor.. moreover in the current implementation it is unlikely that the properties will contain the modified interface names property, so we will avoid the map cloning in most cases.

In any case if you want I can double check that it is always applied to mutable map and remove the map cloning

Reply:
Oh... I didn't know that. TIL

I think it's ok as-is then 👍